src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1600,25 +1716,28 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
         &self,
         place: &'d Place<'tcx>,
         is_local_mutation_allowed: LocalMutationIsAllowed,
-    ) -> Result<(), &'d Place<'tcx>> {
+    ) -> Result<(&'d Place<'tcx>, LocalMutationIsAllowed), &'d Place<'tcx>> {
```

Comment:
Nit: might be worth making a struct here for this tuple, so we can give names to the fields and write comments explaining what is going on.